godot-macros/src/derive_property.rs

```diff
@@ -0,0 +1,118 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{quote, ToTokens};
+use venial::{Declaration, StructFields};
+
+use crate::{
+    util::{bail, decl_get_info, DeclInfo},
+    ParseResult,
+};
+
+pub fn transform(decl: Declaration) -> ParseResult<TokenStream2> {
+    let DeclInfo {
+        name, name_string, ..
+    } = decl_get_info(&decl);
+
+    let body_get;
+    let body_set;
+    let mut intermediate = quote! {()};
+
+    let enum_ = match decl {
+        Declaration::Enum(e) => e,
+        Declaration::Struct(s) => {
+            return bail!(s.name, "Property can only be derived on enums for now")
+        }
+        _ => unreachable!(),
+    };
+
+    if enum_.variants.is_empty() {
+        body_get = quote! {unreachable!();};
+        body_set = quote! {unreachable!();};
+    } else {
+        let mut matches_get = quote! {};
+        let mut matches_set = quote! {};
+        intermediate = if let Some(attr) = enum_.attributes.iter().find(|attr| {
+            attr.path.len() == 1
+                && attr.get_single_path_segment().map(|path| path.to_string())
+                    == Some("repr".to_string())
```

Comment:
The `attr.path.len() == 1` condition is already checked by `get_single_path_segment()`, so you can omit it.

You can also simplify it a bit:
```suggestion
            attr.get_single_path_segment() == Some(&ident("repr"))
```